src/main/java/sg/ncl/MainController.java

```diff
@@ -3494,6 +3486,514 @@ public String adminEnergy(Model model,
         return "energy_usage";
     }
 
+    private ProjectDetails getProjectDetails(JSONObject jsonObject) {
+        ProjectDetails projectDetails = new ProjectDetails();
+        projectDetails.setId(jsonObject.getInt("id"));
+        projectDetails.setOrganisationType(jsonObject.getString("organisationType"));
+        projectDetails.setOrganisationName(jsonObject.getString("organisationName"));
+        projectDetails.setProjectName(jsonObject.getString("projectName"));
+        projectDetails.setOwner(jsonObject.getString("owner"));
+        try {
+            projectDetails.setZonedDateCreated(getZonedDateTime(jsonObject.get("dateCreated").toString()));
+        } catch (IOException e) {
+            log.warn("Error getting date created {}", e);
+            projectDetails.setDateCreated("");
+        }
+        projectDetails.setEducation(jsonObject.getBoolean("education"));
+        projectDetails.setServiceTool(jsonObject.getBoolean("serviceTool"));
+        projectDetails.setSupportedBy(jsonObject.getString("supportedBy"));
+        JSONArray usages = jsonObject.getJSONArray("projectUsages");
+        for (int i = 0; i < usages.length(); i++) {
+            JSONObject usage = usages.getJSONObject(i);
+            JSONObject usageId = usage.getJSONObject("id");
+            ProjectUsage projectUsage = new ProjectUsage();
+            projectUsage.setId(usageId.getInt("projectDetailsId"));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "projectDetailsId" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)